ui/src/alerting/actions/notifications/endpoints.ts

```diff
@@ -56,10 +72,107 @@ export const createEndpoint = (data: NotificationEndpoint) => async (
     throw new Error(resp.data.message)
   }
 
-  const endpoint = (resp.data as unknown) as NotificationEndpoint
+  dispatch({
+    type: 'SET_ENDPOINT',
+    endpoint: resp.data,
+  })
+}
+
+export const updateEndpoint = (
+  endpointID: string,
+  endpoint: Partial<NotificationEndpoint>
+) => async (dispatch: Dispatch<Action | NotificationAction>) => {
+  const resp = await api.patchNotificationEndpoint({
```

Comment:
I know it's a huge pain, but it would be nice to track the loading states for each of these thunks. I think the UX can be pretty jarring otherwise.

Reply:
Or I guess that would also require the various label components to be loading state aware too.... prolly too much to worry about right now? blarg